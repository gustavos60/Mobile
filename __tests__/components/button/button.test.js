import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Button from '../../../src/components/button';

const mockedProps = {
  label: 'Butao',
};

describe('Button tests', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<Button {...mockedProps} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should show activity indicator when loading', () => {
    const wrapper = shallow(<Button {...mockedProps} loading />);
    const View = wrapper.find('ActivityIndicator').first();
    expect(View.exists()).toBeTruthy();
  });

  it('should hide activity indicator when not loading', () => {
    const wrapper = shallow(<Button {...mockedProps} loading={false} />);
    const View = wrapper.find('ActivityIndicator').first();
    expect(View.exists()).toBeFalsy();
  });
});
