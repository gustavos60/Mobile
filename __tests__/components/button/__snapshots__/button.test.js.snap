// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#123456",
      "borderRadius": 30,
      "height": 55,
      "opacity": 1,
      "width": 150,
    }
  }
  testID=""
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
          },
          null,
        ]
      }
    >
      Butao
    </Text>
  </View>
</View>
`;
